core/src/apps/management/change_pin.py

```diff
@@ -26,6 +27,10 @@ async def change_pin(msg: ChangePin) -> Success:
     # confirm that user wants to change the pin
     await _require_confirm_change_pin(msg)
 
+    # Do not allow to remove the PIN if the wipe code is set.
+    if msg.remove and config.has_wipe_code():
+        await error_wipe_code_exists()
+
```

Comment:
pinging @andrewkozlik for this one: the deepest consideration here is whether we want to:
* show this warning, revealing the existence of wipe code,
* silently remove wipe code as well,
* or behave as before, keeping the wipe code set